src/Raven.Server/Documents/Replication/OutgoingReplicationHandler.cs

```diff
@@ -529,8 +556,17 @@ private void ReadHeaderResponseAndThrowIfUnAuthorized(DocumentsOperationContext
                         throw new InvalidOperationException($"{Destination.FromString()} replied with unknown status {headerResponse.Status}, message:{headerResponse.Message}");
                 }
             }
+
+            return TcpConnectionHeaderMessage.ReplicationTcpVersion;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private bool CanCommunicateWithReplicationProtocol(int headerResponseVersion)
```

Comment:
I don't think this is helpful. You are going to need to check _what properties_ are valid here.

Reply:
This is going to move, this is wrong and not according to what we talked about.